product_docs/docs/efm/4/04_configuring_efm/05_using_vip_addresses.mdx

```diff
@@ -10,13 +10,41 @@ legacyRedirectsGenerated:
 
 <div id="using_vip_addresses" class="registered_link"></div>
 
-Failover Manager uses the `efm_address` script to assign or release a virtual IP address.
+Failover Manager can be used along with a virtual IP address (VIP) for routing requests to the current primary node. 
 
 !!! Note "Cloud provider support and alternatives"
     Virtual IP addresses aren't supported by many cloud providers. In those environments, use another mechanism, such as an elastic IP address on AWS, that can be changed when needed by a fencing or post-promotion script.
 
-!!! Note "Behavior when the agent is stopped"
-    Failover Manager will not drop the Virtual IP address (if used) from the primary node when the agent is stopped. As a convenience for testing, the primary agent will acquire the VIP at startup if the node does not already have it, but otherwise starting and stopping Failover Manager has no effect on the node having the virtual IP address.
+## Behavior during shutdown of an EFM agent
+
+Failover Manager will not drop the virtual IP address from the primary node when the agent for that node shuts down. As a convenience for testing, the primary node's agent will *acquire* the VIP during startup if the node does not already have it, but otherwise starting and stopping Failover Manager has no effect on whether the node holds the virtual IP address. 
+
+This allows you to upgrade and perform maintenance on EFM services without interrupting access to the database.
+
+## Behavior during promotion of a node from standby to primary
+
+The VIP should be initially assigned to the primary node. When EFM detects failure of the primary node, it will release the VIP and then assign it to a standby node as that node is promoted to be the new primary. 
+
+EFM verifies (via the command configured via the [`ping.server.command`](01_cluster_properties.mdx#ping_server_command) cluster property) that the VIP is not currently in use during promotion of a standby, and will skip assigning it to the new primary node if it is already assigned. This behavior can be disabled via the [`check.vip.before.promotion`](01_cluster_properties.mdx#check_vip_before_promotion) cluster property. 
```

Comment:
"...skip assigning it to the new primary node if it is already assigned." It wouldn't already be assigned to a standby node since standbys aren't supposed to have the vip. I think what you're going for is that efm won't promote if it sees that the VIP is reachable. "...during promotion of a standby, and will not promote [etc]."

Reply:
Yeah, I worded that badly. AFAIK, EFM *will* still promote the node, but skip VIP assignment if the VIP is already reachable. Also there's the retry / monitor mentioned below that handles cases where async behavior results in the old primary node hanging onto it longer than expected. I've reworded.